### PR TITLE
Break out browser history functionality into another component

### PR DESCRIPTION
**Pull request by RyanFuentes**
### Change Log
- Separated out the browser push and local storage features into a specialized component.
- Converted stateless component into a stateful one for selected index tracking.
- Added Procore styles into CSS module.
- Moved some code.

### Discussion Topics:
- This is a major breaking change. The default export from this package no longer has the browser history push and local storage features.
- Ripa.RipaControlled is the component that's now exported with the currently expected functionality (history, local storage).

### CODE REVIEW

File: src/Ripa/Ripa.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React, { PropTypes } from 'react';
```
Comment (ryanyogan) [inline src/Ripa/Ripa.jsx]: `src/Ripa/Ripa`  avoid namespace clashing, perhaps this is a `Root.jsx` or some other convention that should be establish.

File: src/Ripa/Ripa.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import styles from './styles/default.scss';
+
+class Ripa extends React.Component {
+
+  constructor(props) {
+    super(props);
+    if (
```
Comment (ryanyogan) [inline src/Ripa/Ripa.jsx]: Constructor is not intended for logic, perhaps this should be in the `componentWillMount` lifecycle?  The constructor is for initial state, etc...

File: src/Ripa/Ripa.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    return (
+      <div className={styles.container}>
+        <h1>{title}</h1>
+        {tabs}
+      </div>
+    );
+  }
+}
+
+Ripa.propTypes = {
```
Comment (ryanyogan) [inline src/Ripa/Ripa.jsx]: If you want to make things a little more declarative up front:

`const { arrayOf, shape, string, func, number } = PropTypes;` React.PropTypes is not required as it has already been imported and de-structured out of the React object.

File: src/RipaControlled/RipaControlled.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    <Ripa
+      {...props}
+      onChange={onRipaChange(scope, onChange)}
+      initialSelectedKey={getCurrentViewFromBrowser(scope)}
+    />
+  );
+};
+
+const onRipaLoad = (scope) => {
+  const view = localStorage.getItem(`${scope}/VIEW`);
```
Comment (ryanyogan) [inline src/RipaControlled/RipaControlled.jsx]: `const view = Promise.resolve(localStorage.getItem(`${scope}/VIEW`));`

That is blocking code, most likely want to add some form of `asyncronicity` to it.

File: src/RipaControlled/RipaControlled.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  if (view) {
+    const url = window.location.href.split('?');
+    const params = qs.parse(url[1]);
```
Comment (ryanyogan) [inline src/RipaControlled/RipaControlled.jsx]: If `url` resolves to an empty array, `url[1]` //--> Cannot find [1] of undefined

File: src/RipaControlled/RipaControlled.jsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      .pushState(
+        history.state,
+        '',
+        `${url[0]}?${qs.stringify(newParams, { arrayFormat: 'brackets' })}`);
+  }
+};
+
+const onRipaChange = (scope, handler) => (k, v, index) => {
+  const url = window.location.href.split('?');
+  const params = qs.parse(url[1]);
```
Comment (ryanyogan) [inline src/RipaControlled/RipaControlled.jsx]: Same as previous comment

File: src/RipaControlled/RipaControlled.jsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      .pushState(
+        history.state,
+        '',
+        `${url[0]}?${qs.stringify(newParams, { arrayFormat: 'brackets' })}`);
+    handler(k, v, index);
+  }
+};
+
+const getCurrentViewFromBrowser = (scope) => {
+  const params = qs.parse(window.location.search.substr(1));
```
Comment (ryanyogan) [inline src/RipaControlled/RipaControlled.jsx]: I am seeing `localStorage` and `parsing` happen often, perhaps push this logic to its own function, allowing for memorization and better error catching.

File: src/RipaControlled/RipaControlled.jsx (hunk 5 above)
Comment (RyanFuentes) [inline src/RipaControlled/RipaControlled.jsx]: qs.parse(undefined) will resolve to an empty object.

File: src/RipaControlled/RipaControlled.jsx (hunk 7 above)
Comment (RyanFuentes) [inline src/RipaControlled/RipaControlled.jsx]: I think any performance gains from this would be negligible.

**Pull request merged**